lib/client/api.go

```diff
@@ -1439,50 +1440,31 @@ func (tc *TeleportClient) Play(ctx context.Context, namespace, sessionID string)
 		}
 		defer term.Restore(0, state)
 	}
-	player := newSessionPlayer(sessionEvents, stream)
-	// keys:
-	const (
-		keyCtrlC = 3
-		keyCtrlD = 4
-		keySpace = 32
-		keyLeft  = 68
-		keyRight = 67
-		keyUp    = 65
-		keyDown  = 66
-	)
-	// playback control goroutine
-	go func() {
-		defer player.Stop()
-		key := make([]byte, 1)
-		for {
-			_, err = os.Stdin.Read(key)
-			if err != nil {
-				return
-			}
-			switch key[0] {
-			// Ctrl+C or Ctrl+D
-			case keyCtrlC, keyCtrlD:
-				return
-			// Space key
-			case keySpace:
-				player.TogglePause()
-			// <- arrow
-			case keyLeft, keyDown:
-				player.Rewind()
-			// -> arrow
-			case keyRight, keyUp:
-				player.Forward()
-			}
-		}
-	}()
-
-	// player starts playing in its own goroutine
-	player.Play()
+	return playSession(sessionEvents, stream)
+}
 
-	// wait for keypresses loop to end
-	<-player.stopC
-	fmt.Println("\n\nend of session playback")
-	return trace.Wrap(err)
+// PlayFile plays the recorded session from a tar file
+func (tc *TeleportClient) PlayFile(ctx context.Context, tarFile io.Reader, sid string) error {
+	var sessionEvents []events.EventFields
+	var stream []byte
+	protoReader := events.NewProtoReader(tarFile)
+	w, err := events.WriteForPlayback(ctx, session.ID(sid), protoReader, os.TempDir())
```

Comment:
nit: so my idea was that if the client would treat the playback artifacts as a black box and provide just the directory for the player:
```go
playbackDir, err := ioutil.TempDir("", "playback")
// handle err
defer os.Remove(playbackDir) // don't care about specific contents
w, err := events.WriteForPlayback(ctx, session.ID(sid), protoReader, playbackDir)
```
then `PlaybackWriter` would not need to expose anything besides these two APIs: `SessionEvents` and `SessionChunks`.

Reply:
oh i see, will implement